binding.cc

```diff
@@ -79,6 +91,33 @@ Buffer<BYTE> CertToBuffer(Env env, PCCERT_CONTEXT cert, LPCWSTR password, DWORD
   return outbuf;
 }
 
+// Export a X.509 certificate (which does not include a key) as a PEM string.
+String CertToPEMBuffer(Env env, PCCERT_CONTEXT cert) {
+  const unsigned char* pbCertEncoded = reinterpret_cast<const unsigned char*>(cert->pbCertEncoded);
+  X509* x509cert = d2i_X509(nullptr, &pbCertEncoded, cert->cbCertEncoded);
+  if (!x509cert) {
+    ThrowOpenSSLError(env, "d2i_X509()");
+  }
+  Cleanup cleanup_cert([&]() { X509_free(x509cert); });
+
+  BIO* bio = BIO_new(BIO_s_mem());
+  if (!bio) {
+    ThrowOpenSSLError(env, "BIO_new()");
+  }
+  Cleanup cleanup_bio([&]() { BIO_free(bio); });
```

Comment:
I am not sure how this is supposed to work. I suppose it calls the lambda, that would clear the BIO when cleanup_bio is out of the lexical scope right?

Reply:
Yup, exactly that.